clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1138,7 +1138,7 @@ public void onFailure(RuntimeException e) {
                                             // however, then the session timeout may expire before we can rejoin.
                                             heartbeat.receiveHeartbeat();
                                         } else if (e instanceof FencedInstanceIdException) {
-                                            log.error("Caught fenced group.instance.id error in heartbeat thread", e);
+                                            log.error("Caught fenced group.instance.id {} error in heartbeat thread", groupInstanceId, e);
```

Comment:
Note if you have N `{}` in the string with N+1 parameters in the end, the last one would not be automatically translated to the error stack trace but would be dropped. Log4j is not that smart..

I think here since we know exactly `e instance of FencedInstanceIdException` we can just remove `e` in the parameter.

Reply:
Actually slf4j is that smart :wink:. https://www.slf4j.org/faq.html#paramException

Reply:
But I agree we should avoid printing the stacktrace when we know the error.

Reply:
Agree, will remove!

Reply:
I must have seen an old version of log4j :P